### ISSUE TITLE
Update markers as the location changes

### ISSUE THREAD

**Issue by aadimator** [appoly/ARCore-Location#30]
I want to create a real-time location tracker where users will be able to track each other. How to update the Location Markers in the AR world when the location of the user changes? I've tried removing that location node using `locationScene.mLocationMarkers.remove(index)` and adding the updated version to the `mLocationMarkers` but it's not working. When I close out of the ArActivity and re-launch it, it uses the updated location but not before. Any ideas on how I might be able to approach this problem?

**Comment by morhpeus87** [comment]
Hi,
`locationScene.mLocationMarkers.remove(index)` only removes the LocationMarker object from the arraylist, but the LocationNode is still part of the scene. Before calling `locationScene.mLocationMarkers.remove(index)` I think you should add this piece of code (taken from lines 268-271 from LocationScene class)
`mLocationMarkers.get(i).anchorNode.getAnchor().detach();
                        mLocationMarkers.get(i).anchorNode.setAnchor(null);
                        mLocationMarkers.get(i).anchorNode.setEnabled(false);
                        mLocationMarkers.get(i).anchorNode = null;`
I've prooved it and its working. I had a similar task in which I had to change the rendered markers by choosing the desidered search radius.

Hopefully maybe some of the developers can tell us if this solution is correct or not.

**Comment by johnwedgbury** [comment]
@morhpeus87 's mentioned method is currently the way that we do it - as unsightly as it is. If anyone has a better solution however, please feel free to contribute.

**Comment by aadimator** [comment]
Thanks for your response @morhpeus87 I've tried this method and it seems to be working fine, for removing that marker from the scene. But the problem is I want to add the updated marker to the scene as well. I'm using `locationScene.mLocationMarkers.add(updatedMarker);` but it doesn't seem to have any effect.

**Comment by aadimator** [comment]
I've found one solution to this problem. When I add the marker to `mLocationMarkers`, in the next line, I'm calling `locationScene.refreshAnchors();` This seems to be working fine for now. Is this the right way to approach this problem or is there some better, more efficient way?

**Comment by SheikhZayed** [comment]
@morhpeus87 how did you achieve to display only the markers in specific radius,could you please post your approach here?
